ocw/lib/EC2.py

```diff
@@ -74,6 +74,73 @@ def list_regions(self):
     def delete_instance(self, instance_id):
         self.ec2_resource().instances.filter(InstanceIds=[instance_id]).terminate()
 
+    def parse_image_name(self, img_name):
+        # openqa-SLES12-SP5-EC2.x86_64-0.9.1-BYOS-Build1.55.raw.xz
+        regex = re.compile(r'''^openqa-SLES
+                        (?P<version>\d+(-SP\d+)?)
+                        -EC2
+                        \.
+                        (?P<arch>[^-]+)
+                        -
+                        (?P<kiwi>\d+\.\d+\.\d+)
+                        -
+                        (?P<flavor>(BYOS|On-Demand))
+                        -Build
+                        (?P<build>\d+\.\d+)
+                        \.raw\.xz
+                        ''', re.RegexFlag.X)
+        m = re.match(regex, img_name)
+        if m:
+            return {
+                    'key': '-'.join([m.group('version'), m.group('flavor'), m.group('arch')]),
+                    'build': "-".join([m.group('kiwi'), m.group('build')]),
+                    }
+
+        # openqa-SLES15-SP2.x86_64-0.9.3-EC2-HVM-Build1.10.raw.xz'
+        # openqa-SLES15-SP2-BYOS.x86_64-0.9.3-EC2-HVM-Build1.10.raw.xz'
+        # openqa-SLES15-SP2.aarch64-0.9.3-EC2-HVM-Build1.49.raw.xz'
+        regex = re.compile(r'''^openqa-SLES
+                        (?P<version>\d+(-SP\d+)?)
+                        (-(?P<type>)[^\.]+)?
+                        \.
+                        (?P<arch>[^-]+)
+                        -
+                        (?P<kiwi>\d+\.\d+\.\d+)
+                        -
+                        (?P<flavor>EC2[-\w]+)
+                        -Build
+                        (?P<build>\d+\.\d+)
+                        \.raw\.xz
+                        ''', re.RegexFlag.X)
+        m = re.match(regex, img_name)
+        if m:
+            return {
+                    'key': '-'.join([m.group('version'), m.group('flavor'), m.group('type') or '', m.group('arch')]),
```

Comment:
we have 'type' only here

Reply:
in regex above `BYOS` is called `flavor`  and here same thing is `type` . this is confusing

Reply:
That is because the "flavor" is splitted in two parts.
One is `EC2-HVM` and the other is `BYOS` or not.

It's more clear for azure images. There we have a Flavor `Azure-Standard`, `Azure-Basic` and each with and without `BYOS`. 

In the past, without `BYOS` was called `On-Demand`.